src/io.c

```diff
@@ -1059,18 +1066,22 @@ void *io_thread_rfcomm(void *arg) {
 	uint8_t mic_gain = t->rfcomm.sco->sco.mic_gain;
 	uint8_t spk_gain = t->rfcomm.sco->sco.spk_gain;
 	char buffer[64];
+	struct at_command at;
 
 	struct pollfd pfds[] = {
 		{ t->event_fd, POLLIN, 0 },
 		{ t->bt_fd, POLLIN, 0 },
 	};
 
+	/* HSP only supports CVSD */
+	if (t->profile == BLUETOOTH_PROFILE_HSP_HS || t->profile == BLUETOOTH_PROFILE_HSP_AG)
+		t->rfcomm.sco->sco.codec = TRANSPORT_SCO_CODEC_CVSD;
```

Comment:
This value should be set during the transport creation, since it is the default (and the only) value for HSP.